phoenix-protocol/src/main/MetaDataService.proto

```diff
@@ -75,7 +76,8 @@ message MetaDataResponse {
   repeated SharedTableState sharedTablesToDelete = 9;
   optional PSchema schema = 10;
   optional int64 autoPartitionNum = 11;
-  optional int32 viewIndexId = 12;
```

Comment:
@twdsilva Just for clarification. Do you mean to add a new column like
`optional bytes viewIndexLongId = 13;`
and then retire viewIndexId in the next version. 
This make sense but I only don't understand why the new property cannot be a optional int64?

Reply:
You are correct the new property can be an optional int64.

Reply:
👍 Thanks for quick response

Reply:
@twdsilva @JamesRTaylor 
I just ran into this:
https://developers.google.com/protocol-buffers/docs/proto#updating
```
int32, uint32, int64, uint64, and bool are all compatible – this means you can change a field from one of these types to another without breaking forwards- or backwards-compatibility. If a number is parsed from the wire which doesn't fit in the corresponding type, you will get the same effect as if you had cast the number to that type in C++ (e.g. if a 64-bit number is read as an int32, it will be truncated to 32 bits)
``` 
If I understand it correctly, we can just add the `indexType` column to the .proto wherever we have viewIndexId and change the `viewIndexId` type from `int32` to `int64` without any side effect for old clients and hence we won't need to add `viewIndexLongId` and PHOENIX-4838.
Am I missing anything?

Reply:
From the doc it looks like it should be ok to just change the viewIndexId type from int32 to int64.
We don't have a good way to test that this in a unit test. You could test that this works correctly manually by creating a view with the 4.14 server and client jar and then replacing the server jar with one that has your patch and verifying that you can still run queries against the view.